src/AutoRouteManager.php

```diff
@@ -124,25 +124,51 @@ public function handleDefunctRoutes()
     }
 
     /**
+     * Find an existing route which matches the URI of the given context.
+     *
+     * It is searched within the currently processed collection and the already
+     * persisted routes (using the adapter).
+     */
+    private function findExistingRoute(
+        UriContext $uriContext,
+        UriContextCollection $uriContextCollection
+    ) {
+        $uri = $uriContext->getUri();
+        $existingRoute = null;
+
+        // As the auto route is put in the context after the conflict has been
+        // resolved, we don't need to check if the found auto route is the one
+        // contained in the given context.
+        $existingRoute = $uriContextCollection->getAutoRouteByUri($uri);
+
+        if (is_null($existingRoute)) {
```

Comment:
Consistency,we have tended to use `null === $existingRoute` in other places (same with other uses of `is_null` inthis PR).